src/libraries/System.Private.CoreLib/src/System/IO/TextReader.cs

```diff
@@ -368,9 +376,15 @@ protected override void Dispose(bool disposing)
             [MethodImpl(MethodImplOptions.Synchronized)]
             public override Task<string?> ReadLineAsync() => Task.FromResult(ReadLine());
 
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override ValueTask<string?> ReadLineAsync(CancellationToken cancellationToken) => cancellationToken.IsCancellationRequested ? ValueTask.FromCanceled<string?>(cancellationToken) : new ValueTask<string?>(ReadLine());
```

Comment:
nit: this line is rather too long

```suggestion
            public override ValueTask<string?> ReadLineAsync(CancellationToken cancellationToken)
                => cancellationToken.IsCancellationRequested ? ValueTask.FromCanceled<string?>(cancellationToken) : new ValueTask<string?>(ReadLine());
```